tests/joins_test.go

```diff
@@ -400,3 +402,53 @@ func TestNestedJoins(t *testing.T) {
 		CheckPet(t, *user.Manager.NamedPet, *users2[idx].Manager.NamedPet)
 	}
 }
+
+func TestJoinsPreload_Issue7013(t *testing.T) {
+	manager := &User{Name: "Manager"}
+	DB.Create(manager)
+
+	var userIDs []uint
+	for i := 0; i < 21; i++ {
+		user := &User{Name: fmt.Sprintf("User%d", i), ManagerID: &manager.ID}
+		DB.Create(user)
+		userIDs = append(userIDs, user.ID)
+	}
+
+	var entries []User
+	assert.NotPanics(t, func() {
+		assert.NoError(t,
+			DB.Debug().Preload("Manager.Team").
+				Joins("Manager.Company").
+				Find(&entries).Error)
+	})
+}
+
+func TestJoinsPreload_Issue7013_RelationEmpty(t *testing.T) {
+	home := &Building{Name: "relation_empty"}
+	DB.Create(home)
+
+	var entries []Building
+	assert.NotPanics(t, func() {
+		assert.NoError(t,
+			DB.Debug().Preload("Owner.Furnitures").
+				Joins("Owner.Company").
+				Find(&entries).Error)
+	})
+
+	AssertEqual(t, len(entries), 1)
+	AssertEqual(t, len(entries[0].Owner.Furnitures), 0)
+	AssertEqual(t, entries[0].Owner.Company.ID, 0)
+}
+
+func TestJoinsPreload_Issue7013_NoEntries(t *testing.T) {
+	var entries []User
+	assert.NotPanics(t, func() {
+		assert.NoError(t,
+			DB.Debug().Preload("Manager.Team").
+				Joins("Manager.Company").
+				Where("false").
+				Find(&entries).Error)
+	})
+
+	AssertEqual(t, len(entries), 0)
```

Comment:
We can use `AssertEqual` to describe the results of the actual query, not just the count.

Reply:
in that case I think it's legit to check it's empty but I used it to check content in the other unit test